extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeDashManifestCreator.java

```diff
@@ -235,17 +233,14 @@ public static final class YoutubeDashManifestCreationException extends Exception
      * @param durationSecondsFallback the duration of the video, which will be used if the duration
      *                                could not be extracted from the first sequence
      * @return the manifest generated into a string
-     * @throws YoutubeDashManifestCreationException if something goes wrong when trying to generate
-     *                                              the DASH manifest
      */
     @Nonnull
-    public static String createDashManifestFromOtfStreamingUrl(
+    public static String fromOtfStreamingUrl(
             @Nonnull final String otfBaseStreamingUrl,
             @Nonnull final ItagItem itagItem,
-            final long durationSecondsFallback) throws YoutubeDashManifestCreationException {
-        if (GENERATED_OTF_MANIFESTS.containsKey(otfBaseStreamingUrl)) {
-            return Objects.requireNonNull(GENERATED_OTF_MANIFESTS.get(otfBaseStreamingUrl))
-                    .getSecond();
+            final long durationSecondsFallback) throws CreationException {
+        if (OTF_CACHE.containsKey(otfBaseStreamingUrl)) {
+            return Objects.requireNonNull(OTF_CACHE.get(otfBaseStreamingUrl)).getSecond();
```

Comment:
I think AS would give warnings otherwise?

Reply:
Indeed, but there is one:

<img src="https://user-images.githubusercontent.com/74829229/167675958-50ea1c3f-e9b7-4a2f-ab39-bb21f9816f4e.png" width="600px" alt="Warning IDEA nullity get ManifestCreatorCache">

So this will be kept as it is.